source/application/AMPIC887DriverServer/AMPIC887ConsoleApplication.cpp

```diff
@@ -70,23 +66,507 @@ void AMPIC887ConsoleApplication::onActiveControllerChangeIssued(const QString &c
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetPivotPointCommandIssued(const AMPIC887AxisMap<double> &pivotPoints)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setPivotPoints(pivotPoints)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetRecorderTriggerCommandIssued(AMGCS2::DataRecordTrigger recorderTrigger)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setRecordTrigger(recorderTrigger)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetRecordConfigCommandIssued(const QHash<int, AMPIC887DataRecorderConfiguration> &recordConfigs)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setRecordConfigs(recordConfigs)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetServoModeCommandIssued(bool servoMode)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setServoMode(servoMode)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetSoftLimitStatesCommandIssued(const AMPIC887AxisMap<bool> &softLimitStates)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setSoftLimitStatuses(softLimitStates)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetStepSizeCommandIssued(const AMPIC887AxisMap<double> &stepSizes)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setStepSizes(stepSizes)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onSetSystemVelocityCommandIssued(double systemVelocity)
+{
+	AMPIC887Controller* activeController = controllerCollection_.activeController();
+	if(!activeController) {
+		return;
+	}
+
+	if(!activeController->setSystemVelocity(systemVelocity)) {
+		consoleInputHandler_->writeLineToStandardError(activeController->lastError());
+	}
+}
+
+void AMPIC887ConsoleApplication::onMoveCommandIssued(const AMPIC887AxisMap<double> &axisPositions)
+{
+	if(!controllerCollection_.activeController() && !controllerCollection_.activeController()->isInitialized()) {
```

Comment:
This is probably supposed to be an OR?  If you use the style I mentioned earlier then you could just DeMorgan this conditional and it would be fine.